src/widget/wcollapsiblegroupbox.cpp

```diff
@@ -0,0 +1,107 @@
+#include "widget/wcollapsiblegroupbox.h"
+
+#include <QEvent>
+#include <QStyleOption>
+#include <QStylePainter>
+
+#include "moc_wcollapsiblegroupbox.cpp"
+#include "util/assert.h"
+
+void CollapsibleGroupBoxStyle::drawPrimitive(QStyle::PrimitiveElement element,
+        const QStyleOption* pOption,
+        QPainter* pPainter,
+        const QWidget* pWidget) const {
+    const WCollapsibleGroupBox* pCGB = qobject_cast<const WCollapsibleGroupBox*>(pWidget);
+    if (pCGB && pCGB->isCheckable()) {
+        if (element == QStyle::PE_IndicatorCheckBox) {
+            // Draw Down/Right arrow instead of original checkbox
+            QProxyStyle::drawPrimitive(pCGB->isChecked()
+                            ? QStyle::PE_IndicatorArrowDown
+                            : QStyle::PE_IndicatorArrowRight,
+                    pOption,
+                    pPainter,
+                    pWidget);
+            return;
+        } else if (!pCGB->isChecked() && element == QStyle::PE_FrameGroupBox) {
+            // When the box is collapsed, the default style might still draw the
+            // frame, e.g. underneath the title, even though the box is reduced
+            // to its title row, so skip drawing the frame.
+            return;
+        }
+    }
+
+    QProxyStyle::drawPrimitive(element, pOption, pPainter, pWidget);
+}
+
+WCollapsibleGroupBox::WCollapsibleGroupBox(QWidget* pParent)
+        : QGroupBox(pParent),
+          m_minH(-1),
+          m_maxH(10000) {
```

Comment:
Could we use a constant here? Or maybe some `std::numeric_limits`?

Reply:
I tried `std::numeric_limits<int>::quiet_NaN()` earlier but it didn't work as expected, and I was lazy / it was late ; )
Will check again..

Reply:
Ah, that's only available for floating-point types.
so `std::numeric_limits<T>::min` seems to be the only applicable member?

Reply:
Hmm, while it works with `std::numeric_limits<int>::min()`, I'd rather stick with `-1` here:
just in case one of the `style()->subControlRect()` queries returns a QRect with an invalid height (-1) we have the chance to set it on the next call.

Wdyt?

Reply:
imo fine as long as you refactor it into a `static constexpr` constant. no magic numbers and all that u know.